metricbeat/module/elasticsearch/cluster_stats/data.go

```diff
@@ -0,0 +1,92 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cluster_stats
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/elasticsearch"
+)
+
+var (
+	schema = s.Schema{
+		"status": c.Str("status"),
+		"nodes": c.Dict("nodes", s.Schema{
+			"total":  c.Int("count.total"),
+			"master": c.Int("count.master"),
+			"data":   c.Int("count.data"),
+		}),
+		"indices": c.Dict("indices", s.Schema{
+			"total": c.Int("count"),
+			"shards": c.Dict("shards", s.Schema{
+				"total":     c.Int("total"),
+				"primaries": c.Int("primaries"),
+			}),
+			"segments": c.Dict("segments", s.Schema{
+				"total": c.Int("count"),
+			}),
+			"docs": c.Dict("docs", s.Schema{
+				"total":   c.Int("count"),
+				"deleted": c.Int("deleted"),
+			}),
+			"store": c.Dict("store", s.Schema{
+				"size": s.Object{
+					"bytes": c.Int("size_in_bytes"),
+				},
+			}),
+			"fielddata": c.Dict("fielddata", s.Schema{
+				"memory_size": s.Object{
+					"bytes": c.Int("memory_size_in_bytes"),
+				},
+			}),
+		}),
+	}
+)
+
+func eventMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte) error {
+	var data map[string]interface{}
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		r.Error(err)
+		return err
+	}
+
+	metricSetFields, err := schema.Apply(data)
+	if err != nil {
+		r.Error(err)
+		return err
+	}
+
+	var event mb.Event
+	event.RootFields = common.MapStr{}
+	event.RootFields.Put("service.name", "elasticsearch")
+
+	event.ModuleFields = common.MapStr{}
+	event.ModuleFields.Put("cluster.name", info.ClusterName) // TODO cluster_name from cluster stats response could be used here?
+	event.ModuleFields.Put("cluster.id", info.ClusterID)     // TODO: Can we get this returned in cluster stats response itself?
```

Comment:
++

I think both values are a must to be returned y cluster stats API

Reply:
Created https://github.com/elastic/elasticsearch/issues/32205 to request cluster UUID from cluster stats API itself.

Reply:
... and put up PR as well: https://github.com/elastic/elasticsearch/pull/32206

Reply:
Keep in mind that older versions of ES will not have it in, so in these cases we need to skip it.

Reply:
Good point! Should we skip it or make the extra API call in that case (I realize that complicates the logic/organization of the code a fair bit)?

Reply:
Lets skip it to not add more complexity. New versions, new features :-)

Reply:
Fixed in ed74319b4.